### PR TITLE
Fix: Phone number hint not working issue

### PR DESCRIPTION
**Pull request by sahilmehra98**
1. Fixed "java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/android/gms/auth/api/credentials/HintRequest$Builder;" error.
2. Upgraded Gradle

### CODE REVIEW

**Comment by sahilmehra98** [comment]
This PR solves issues #230, #229, #226, #225

**Comment by sahilmehra98** [comment]
@jaumard Please review this. If all okay, please merge this as a lot of people are facing this issue.

**Comment by AryanBisht-03** [comment]
@jaumard I m facing similar issue, please merge this one so that I can also start working on my project

**Comment by AryanBisht-03** [comment]
@sahilmehra98 I am in need of this fix, can I implement this fix directly in my flutter project, as I need it urgently.?

**Comment by jaumard** [comment]
with your changes I'm not able to run the project anymore, no issue with master. Do it work on your side? 

Here is what I get when I try to run the example project: 
```
Running Gradle task 'assembleDebug'...
Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/03 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/03 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/03 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:mergeDebugAssets'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Failed to transform annotation-experimental-1.4.0.aar (androidx.annotation:annotation-experimental:1.4.0) to match attributes {artifactType=android-assets, org.gradle.category=library, org.gradle.dependency.bundling=external, org.gradle.libraryelements=aar, org.gradle.status=release, org.gradle.usage=java-runtime}.
      > Execution failed for JetifyTransform: /Users/jaumard/.gradle/caches/modules-2/files-2.1/androidx.annotation/annotation-experimental/1.4.0/c7a2038d831d0cfbfb9a440890b287c2e839ecf/annotation-experimental-1.4.0.aar.
         > Failed to transform '/Users/jaumard/.gradle/caches/modules-2/files-2.1/androidx.annotation/annotation-experimental/1.4.0/c7a2038d831d0cfbfb9a440890b287c2e839ecf/annotation-experimental-1.4.0.aar' using Jetifier. Reason: IllegalArgumentException, message: Unsupported class file major version 61. (Run with --stacktrace for more details.)
           Suggestions:
            - Check out existing issues at https://issuetracker.google.com/issues?q=componentid:460323&s=modified_time:desc, it's possible that this issue has already been filed there.
            - If this issue has not been filed, please report it at https://issuetracker.google.com/issues/new?component=460323 (run with --stacktrace and provide a stack trace if possible).

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 8s

Exception: Gradle task assembleDebug failed with exit code 1
```

File: example/.flutter-plugins-dependencies [hunk 1]
```diff
@@ -0,0 +1 @@
+{"info":"This is a generated file; do not edit or check into version control.","plugins":{"ios":[{"name":"sms_autofill","path":"/Users/sahilmehra/Work/fork/sms_autofill/","native_build":true,"dependencies":[]}],"android":[{"name":"sms_autofill","path":"/Users/sahilmehra/Work/fork/sms_autofill/","native_build":true,"dependencies":[]}],"macos":[],"linux":[],"windows":[],"web":[]},"dependencyGraph":[{"name":"sms_autofill","dependencies":[]}],"date_created":"2024-06-07 02:15:30.350562","version":"3.22.1"}
```
Comment (jaumard) [inline example/.flutter-plugins-dependencies:1]: can you resolve conflicts and remove this file please?

**Comment by sahilmehra98** [comment]
> Failed to transform annotation-experimental-1.4.0.aar

This has been fixed

**Comment by jaumard** [comment]
thanks @sahilmehra98 ! Works now on my side, please fix the conflicts so I can merge and deploy this.

File: example/.flutter-plugins-dependencies (hunk 1 above)
Comment (sahilmehra98) [reply example/.flutter-plugins-dependencies]: Resolved

**Comment by sahilmehra98** [comment]
> thanks @sahilmehra98 ! Works now on my side, please fix the conflicts so I can merge and deploy this.

Resolved

**Comment by jaumard** [comment]
@sahilmehra98 thanks a lot, I'll push that to pub today.

**Pull request merged**